types/invoker_test.go

```diff
@@ -0,0 +1,178 @@
+package types
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"sync"
+	"testing"
+)
+
+func TestInvoker_Invoke(t *testing.T) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		function := strings.Split(r.URL.Path, "/")[2]
+
+		switch function {
```

Comment:
Is this acting as the gateway?

Reply:
Technically it is the functions endpoint, that is proxied via the Gateway. And the of watchdog also allows for none 200 or 500 (503) responses

Reply:
I'm a bit confused by the naming.

Is this HTTP server where functions will be invoked? It seems like it may be something like a "fakeGateway" vs "srv"?

Reply:
I can rename it to fakeGateway, if that is more clear to you. The issue however is that OpenFaaS allows direct invocations of functions or over the gateway, as far as I'm aware.

Reply:
Done in 435c979